### PR TITLE
Add rejectEmptyResponse example

### PR DESCRIPTION
**Pull request by jlprat**
Issue: #218 
Add example about `rejectEmptyResponse` for Java.
Add warning about completing a route with an Optional and its non expected output.

### CODE REVIEW

**Comment by jlprat** [comment]
PLS BUILD

File: docs/src/main/paradox/java/http/routing-dsl/directives/misc-directives/rejectEmptyResponse.md [hunk 1]
```diff
@@ -5,10 +5,17 @@
...
+The `rejectEmptyResponse` directive is can be used whenever responding with an empty `Entity` should be a rejection 
+instead of an `OK` response. One easy example where this can be used is when the entity is a `String` and the 
+empty `String` should be handled as `404 Not Found`.
+
+## Warning
+
+Java's `Optional<T>` does not behave like Scala's `Option[T]` as it's rendered as an own `Entity` instead of being 
+rendered according the parametrized `T` type, if present.
+
+This implies that, a non-empty `Optional` is rendered like `{"present":true}` and an empty one like `{"present":false}`.
```
Comment (ktoso) [inline docs/src/main/paradox/java/http/routing-dsl/directives/misc-directives/rejectEmptyResponse.md]: I think we rather need to fix the directive. Keeping open for now.

Comment (jlprat) [inline docs/src/main/paradox/java/http/routing-dsl/directives/misc-directives/rejectEmptyResponse.md]: That's why I put these changes in a separate PR :)

**Comment by jlprat** [comment]
For reference: https://github.com/akka/akka-http/issues/559

**Comment by ktoso** [comment]
Let's close this, a bit too fuzzy.

File: docs/src/main/paradox/java/http/routing-dsl/directives/misc-directives/rejectEmptyResponse.md [hunk 2]
```diff
@@ -5,10 +5,17 @@
...
```
Comment (jlprat) [inline docs/src/main/paradox/java/http/routing-dsl/directives/misc-directives/rejectEmptyResponse.md]: Yes, it does
